### PR TITLE
Doc: Restructure source files for pipeline configuration

### PR DESCRIPTION
**Pull request by karenzone**
Break monolithic source file into smaller files in preparation for rewrite and migration to new docs system. 

Note that this PR only restructures source files. Changes to content will come later in a follow-up PR to facilitate easier and more thorough review. Replaces #13878 that combined both restructured files and content changes.  Too much to review.  

**PREVIEW:** https://logstash_13990.docs-preview.app.elstc.co/diff

### CODE REVIEW

**Comment by karenzone** [comment]
@jsvd As requested... splitting restructure and content changes into separate PRs.

**Review by jsvd — APPROVED** [review]
LGTM, I see the PR is in draft mode but seems it can be merged, right?

File: docs/index.asciidoc [hunk 1]
```diff
@@ -75,7 +75,7 @@ include::static/shutdown.asciidoc[]
 include::static/upgrading.asciidoc[]
 
 // Configuring Logstash
-include::static/configuration.asciidoc[]
+include::static/pipeline-configuration.asciidoc[]
```
Comment (karenzone) [inline docs/index.asciidoc:79]: Note to self @karenzone: Verified that all existing content is accounted for in source file restructure.

**Comment by karenzone** [comment]
@logstashmachine backport 8.2

**Comment by karenzone** [comment]
@logstashmachine backport 8.1

**Comment by karenzone** [comment]
@logstashmachine backport 8.0

**Pull request merged**